src/mongocxx/test/spec/unified_test_format.cpp

```diff
@@ -423,6 +420,42 @@ std::vector<std::string> versions_to_string(schema_versions_t versions) {
     return out;
 }
 
+std::vector<document::view> array_elements_to_documents(array::view array) {
+    // no implicit conversion from 'bsoncxx::array::view' to 'bsoncxx::document::view'
+    auto docs = std::vector<document::view>{};
+    auto arr_to_doc = [](const array::element& doc) { return doc.get_document().value; };
+
+    std::transform(std::begin(array), std::end(array), std::back_inserter(docs), arr_to_doc);
+    return docs;
+}
+
+bool add_data_to_collection(const array::element& data) {
+    auto db_name = data["databaseName"].get_string().value;
+    auto& map = get_entity_map();
+    auto& db = map.get_database_by_name(db_name);
+
+    auto wc = write_concern{};
+    wc.majority(std::chrono::milliseconds{0});
+
+    auto coll_name = data["collectionName"].get_string().value;
+
+    if (db.has_collection(coll_name))
+        db[coll_name].drop();
+
+    auto coll = db.create_collection(coll_name, {}, wc);
+
+    auto to_insert = array_elements_to_documents(data["documents"].get_array().value);
+    return to_insert.empty() || coll.insert_many(to_insert)->result().inserted_count() != 0;
+}
+
+void load_initial_data(document::view test) {
+    if (!test["initialData"])
+        return;
+
+    auto data = test["initialData"].get_array().value;
+    std::all_of(std::begin(data), std::end(data), add_data_to_collection);
```

Comment:
This is an odd way to use std::all_of(), particularly since you're not handling the failure case.  As soon as one element isn't inserted to the collection, the loop is going to stop.  I would think failure should be dealt with somehow, but if failure is permissible then surely skipping the failures and continuing on with the rest would be the better regression, no?  In that case, maybe use std::for_each() without that ignored bool return?

Reply:
Actually, just noticed std::for_each() is C++17 which you probably can't force at this point in time. Still the failure possibility should be addressed.

Reply:
Ah, this should've been wrapped in a `REQUIRE` macro since no failures are allowed. I ended up just making `add_data_to_collection` return void and asserting the inserted count in that function to capture more info for debugging.